syntax = "proto3";

package obby.sse;

option go_package = "github.com/obby/sse-hub/proto/generated";

service SSEService {
  // Publish an event
  rpc Publish(PublishRequest) returns (PublishResponse);

  // Register client and stream messages
  rpc RegisterClient(ClientRequest) returns (stream SSEMessage);
}

message PublishRequest {
  string event = 1;
  string topic = 2;
  string data = 3;
}

message PublishResponse {
  bool success = 1;
  int32 clients = 2;
}

message ClientRequest {
  repeated string topics = 1;
}

message SSEMessage {
  string event = 1;
  string topic = 2;
  string data = 3;
}

